Likes: reimplement non-admin portions without jQuery

sgomes opened the pull request:
This PR reimplements the non-admin portions of Likes (namely `queuehandler.js`) without jQuery.

It also removes what appears to be deprecated functionality, in the form of a `slim-likes-widget` option. There appears to be no reference to this anywhere else in Jetpack or wpcom, which makes me think this may have been an early version of a different "slim" option that does exist today (`wpl-slim-likebox`) but isn't directly referenced in Jetpack.

`post-count.js` and `post-count-jetpack.js` are not touched in this PR, as they're only loaded in the context of `wp-admin`, which is not as much of a performance concern and thus out of scope for this PR.

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
* Reimplement `queuehandler.js` without jQuery
* Remove apparently deprecated `slim-likes-widget` option
* Remove `jquery.jetpack-resize.js`, as the apparently deprecated `slim-likes-widget` option was the only thing that made use of it
* Change Jetpack plugin module linting to be ES2019-friendly
* Remove `queuehandler.js` and deleted `jquery.jetpack-resize.js` from ESLint exclusion list

#### Jetpack product discussion
None.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
- Ensure that the Likes and Comment Likes functionality continues to work correctly in Jetpack sites.
Please try different sites, as some of them use different code-paths; for example, some sites have a link that presents a popup with "other gravatars".
- Ensure that the Likes and Comment Likes settings pages continue to work correctly. Nothing is expected to have changed for these.

sgomes committed:
Likes: reimplement non-admin portions without jQuery

sgomes commented on projects/plugins/jetpack/modules/likes/.eslintrc.js:
```diff
@@ -0,0 +1,46 @@
+// eslint-disable-next-line
```
Note: the commit checks were still trying to lint this file, even with it added to the exclude list, which forced me to add this line.

anomiex replied:
The exclude list specifies which files have existing errors that should be ignored, but our tooling still checks *changed* lines. So it doesn't really work to add a new file to the exclude list.

Instead of disabling everything, though, better to disable just the one warning that's being triggered:
```suggestion
// eslint-disable-next-line no-undef
```

Although even better IMO would be to only override what you need to change from the parent config instead of overriding the entire config. Maybe clean up obsolete junk from `projects/plugins/jetpack/modules/.eslintrc.js`, like do we really need to restrict modules to es5 anymore?

sgomes replied:
Good points, @anomiex 👍

I'll give `projects/plugins/jetpack/modules/.eslintrc.js` a cleanup, since even if we still have any ES5 files around, we shouldn't need to check that they're valid ES5 any longer.

sgomes replied:
@anomiex Done. The Jetpack plugin modules lint configuration has been updated to not require ES5 any more.

anomiex replied:
Great, thanks for doing that!

sgomes committed:
Move ESLint configuration to module root

samiff committed:
Merge branch 'master' into update/remove-jquery-from-likes

kraftbj commented on projects/plugins/jetpack/modules/.eslintrc.js:
```diff
@@ -1,39 +1,31 @@
+		es6: true,
 	},
 	globals: {
-		_: false,
```
these three lines need to remain.

samiff replied:
Internal ref: p1636751680290600-slack-CBG1CP4EN

samiff committed:
Try fixing ESLint check

samiff committed:
More linter fixes

samiff committed:
More linter fixing

`warning  'jQuery' is already defined as a built-in global variable no-redeclare`

samiff reviewed (approved):
Looks good 👍   Tested on local JP dev env & a simple sandbox site.

A couple minor changes were committed to appease the linter checks.

sgomes commented:
Thank you for the reviews and fixes, @samiff and @kraftbj 🙏

sgomes commented:
r235036
